api/commissionApi.py

```diff
@@ -0,0 +1,185 @@
+import u2py
+from flask import Flask, Response
+import json
+
+from flask_cors import CORS
+
+app = Flask(__name__)
+app.config['SECRET_KEY'] = 'thisisthesercretkey'
+CORS(app)
+
+
+def convert_field_formats(field_data, field_format, format):
+    data = u2py.DynArray()
+    data.insert(1, 0, 0, field_data)
+    if format == 'external':
+        formatted_data = str(data.extract(1).oconv(field_format))
+    else:
+        formatted_data = str(data.extract(1).iconv(field_format))
+    return formatted_data
+
+
+def check_existing_record(file_object, record_ID):
+    try:
+        file_object.read(record_ID)
+        return True
+    except u2py.U2Error:
+        return False
+
+
+def calculate_xmul(trans_data):
+    transaction_type = trans_data['TRAN.TYPE']
+    trans_sub_type = trans_data['TRAN.SUB.TYPE']
+    if transaction_type == 'VSAL':
+        if trans_sub_type == 'CF.CANCEL':
+            xmul = 1
+        else:
+            xmul = -1
+    else:
+        xmul = 1
+    return xmul
+
+
+def calculate_amount(trans_data):
+    amount_list = []
+    comm_sale_amount = [float(key['CommSaleAmt']) for key in trans_data['ITEM_MV']]
+    quantity_list = [int(key['MKUP.STORE.QTY']) for key in trans_data['MKUP.STORE.QTY_MV']]
+    xmul = calculate_xmul(trans_data)
+    for i in range(len(comm_sale_amount)):
+        amount = comm_sale_amount[i]
+        if amount != 0:
+            quantity = quantity_list[i] * xmul
+            amount = convert_field_formats(amount * quantity * 100, 'MD2', 'external')
+            amount_list.append(amount)
+        else:
+            amount_list.append(' ')
+    return amount_list
+
+
+def calculate_commission_amount(trans_data):
+    xmul = calculate_xmul(trans_data)
+    commission_amount = 0
+    empId = [key['CommEmplId'] for key in trans_data['ITEM_MV']]
+    comm_sale_amount = [float(key['CommSaleAmt']) for key in trans_data['ITEM_MV']]
+    quantity_list = [int(key['MKUP.STORE.QTY']) for key in trans_data['MKUP.STORE.QTY_MV']]
+    for i in range(len(empId)):
+        quantity = xmul * quantity_list[i]
+        commission_amount = commission_amount + comm_sale_amount[i] * quantity
+    return commission_amount
+
+
+def set_empsn_values(trans_data, em_file):
+    empsn_list = []
+    comm_emp_id = [key['CommEmplId'] for key in trans_data['ITEM_MV']]
```

Comment:
Avoid abbreviations, easy to type but hard to understand.
Instead of `comm` go with `commission`.
Instead of `emp` go with `employee`.